store/messagestore_sqlite.go

```diff
@@ -23,15 +23,48 @@ func NewPersistentMessageStore(db *sql.DB) *persistentMessageStore {
 
 func (p *persistentMessageStore) Add(message *protobuf.Message) error {
 	id := message.ID()
-	_, err := p.db.Exec(
+
+	tx, err := p.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Exec(
 		`INSERT INTO mvds_messages (id, group_id, timestamp, body)
 		VALUES (?, ?, ?, ?)`,
 		id[:],
 		message.GroupId,
 		message.Timestamp,
 		message.Body,
 	)
-	return err
+
+	if err != nil {
+		_ = tx.Rollback()
+		return err
+	}
+
+	if message.Metadata != nil && len(message.Metadata.Parents) > 0 {
+		query := "INSERT INTO mvds_parents(message, parent) VALUES "
+		vals := make([]interface{}, 0)
```

Comment:
```suggestion
		vals := make([]interface{}, len(message.Metadata.Parents) * 2)
```

Reply:
@adambabik I find that to make the logic a little more annoying

Reply:
It's more efficient but here it does not matter. Alternatively you can go with 
```suggestion
		var vals []interface{}
```
which is equivalent of what you wrote (minus mem allocation in case no appends) and idiomatic way of declaring a slice.